src/node_file.cc

```diff
@@ -1392,18 +1401,20 @@ static void Mkdtemp(const FunctionCallbackInfo<Value>& args) {
 
 void GetStatValues(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
-  double* fields = env->fs_stats_field_array();
-  if (fields == nullptr) {
+  Local<Float64Array> fields = env->fs_stats_field_array();
+
+  if (fields.IsEmpty()) {
     // stat fields contains twice the number of entries because `fs.StatWatcher`
     // needs room to store data for *two* `fs.Stats` instances.
-    fields = new double[2 * 14];
+    Local<ArrayBuffer> ab = ArrayBuffer::New(env->isolate(),
+        new double[2 * 14],
```

Comment:
This is not related to @mrkmarron changes but was like this in upstream. I have [raised the question](https://github.com/nodejs/node/pull/11665/files#r107201164) to check this.